pkg/scalers/gcp/gcp_stackdriver_client.go

```diff
@@ -339,10 +339,13 @@ func getActualProjectID(s *StackDriverClient, projectID string) string {
 // | align delta(3m)
 // | every 3m
 // | group_by [], count(value)
-func (s StackDriverClient) BuildMQLQuery(projectID, resourceType, metric, resourceName, aggregation string) (string, error) {
-	th := defaultTimeHorizon
-	if aggregation != "" {
-		th = aggregationTimeHorizon
+func (s StackDriverClient) BuildMQLQuery(projectID, resourceType, metric, resourceName, aggregation, timeHorizon string) (string, error) {
+	th := timeHorizon
+	if th == "" {
+		th = defaultTimeHorizon
+		if aggregation != "" {
+			th = aggregationTimeHorizon
+		}
```

Comment:
Does it make sense not modifying the aggregation if we don't modify the time horizon? I have no idea about GCP but as this seems a breaking change, if it's something that someone can want to do, I'd just keep them split
```suggestion
	th := timeHorizon
	if th == "" {
		th = defaultTimeHorizon
	}
	if aggregation != "" {
		th = aggregationTimeHorizon
```

Reply:
@JorTurFer But then again the value of timeHorizon would be fixed to 5m in case of aggregation. Though I am not confirm if it is supposed to be 5m only

Reply:
you are totally right, I misunderstood the change